span.go

```diff
@@ -29,6 +29,7 @@ type Span interface {
 type spanImpl struct {
 	tracer     *tracerImpl
 	event      func(SpanEvent)
+	Observer   SpanObserver
```

Comment:
do not export the `SpanObserver` variable as it is not needed and direct external access to it is undesirable.